src/lib/djerba/plugins/wgts/snv_indel/tools.py

```diff
@@ -408,5 +426,8 @@ def write_working_files(self, whizbam_url):
         maf_path_preprocessed = self.preprocess_maf(maf_path, tumour_id)
         maf_path_annotated = self.annotate_maf(maf_path_preprocessed)
         self.run_data_rscript(whizbam_url, maf_path_annotated)
-        self.write_vaf_plot()
+        # Exclude the plot if there are no somatic mutations
+        has_somatic_mutations = self.has_somatic_mutations()
```

Comment:
Simpler to put:
`if self.has_somatic_mutations():`
`    self.write_vaf_plot()`